openpaas-james/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/openpaas/james/common/LinagoraFilterSetMethodContract.scala

```diff
@@ -811,4 +811,513 @@ trait LinagoraFilterSetMethodContract {
+                      |			"accountId": "$generateAccountIdAsString",
+                      |			"update": {
+                      |				"singleton": [{
+                      |					"id": "1",
+                      |					"name": "My first rule",
+                      |					"condition": {
+                      |						"field": "subject",
+                      |						"comparator": "contains",
+                      |						"value": "question"
+                      |					},
+                      |					"action": {
+                      |						"appendIn": {
+                      |							"mailboxIds": ["$generateMailboxIdForUser"]
+                      |						}
+                      |					}
+                      |				}]
+                      |			}
+                      |		}, "c1"]
+                      |	]
+                      |}""".stripMargin
+
+    `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request1)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+
+    val request2 = s"""{
+                      |	"using": ["com:linagora:params:jmap:filter"],
+                      |	"methodCalls": [
+                      |		["Filter/set", {
+                      |			"accountId": "$generateAccountIdAsString",
+                      |			"ifInState": "-1",
+                      |			"update": {
+                      |				"singleton": [{
+                      |					"id": "2",
+                      |					"name": "My second rule",
+                      |					"condition": {
+                      |						"field": "subject",
+                      |						"comparator": "contains",
+                      |						"value": "question"
+                      |					},
+                      |					"action": {
+                      |						"appendIn": {
+                      |							"mailboxIds": ["$generateMailboxIdForUser"]
+                      |						}
+                      |					}
+                      |				}]
+                      |			}
+                      |		}, "c1"]
+                      |	]
+                      |}""".stripMargin
+
+    val response = `given`()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request2)
+    .when()
+      .post()
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |	"sessionState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+         |	"methodResponses": [
+         |		[
+         |			"Filter/set",
+         |			{
+         |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |				"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+         |				"notUpdated": {
+         |					"singleton": {
+         |						"type": "invalidArguments",
```

Comment:
https://jmap.io/spec-core.html#set

`ifInState: If supplied, the string must match the current state; otherwise, the method will be aborted and a stateMismatch error returned.`

Please correct the error handling accordingly.